### ISSUE TITLE
508-defect-3 [COGNITION]: VAOS Community Care - Calendar should announce "AM/PM selected" when users check a box and close the button

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#24628]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

As a screen reader user, I would like to hear a date and AM/PM selected when I check a box in the Community Care calendar date picker. This will help reinforce that I've picked a time slot successfully.

**VFS Point of Contact:** _Trevor_

## User Story or Problem Statement

The calendar date picker cells have buttons to toggle the time picker. When users select a Community Care time slot, we add a visual "AM/PM" status to the button. We should update the aria-label to say "Thursday, May 20th, AM selected" so it reads that additional information to screen readers. Screenshot and code snippet below.

## Acceptance Criteria

- [ ] Community Care date picker button aria-label data updates correctly when an AM/PM slot is checked or unchecked

## Environment

* https://staging.va.gov/health-care/schedule-view-va-appointments/appointments/new-appointment/request-date
* Test user Judy Morrison

## Proposed Solution (if known)

```diff
! We should append the string "AM/PM selected." when users check one or more time slots
! We should remove the string "AM/PM selected." when users uncheck all time slots

<button
  aria-describedby="vaos-calendar-instructions-4"
  aria-expanded="false"
- aria-label="Thursday, May 20th"
+ aria-label="Thursday, May 20th. AM selected."
  class="vaos-calendar__calendar-day-button"
  id="date-cell-2021-05-20"
 type="button"
>
  <div class="vaos-calendar__indicator-bubbles-container">
    <div class="vaos-calendar__indicator-bubble...">
    AM
    </div>
  </div>
20</button>
```

## WCAG or Vendor Guidance (optional)

![Screen Shot 2021-05-12 at 3 04 14 PM](https://user-images.githubusercontent.com/934879/118038327-96867e80-b334-11eb-8e41-2c09612b220e.png)
